### PR TITLE
Fix "length is declared here" issue with TS 4.6

### LINKED ISSUE #947: 'length' is declared here
# Bug report

## Describe the bug

`createStitches` `styled` method requires `length` field to be set. `length` is part of `String` object.

## To Reproduce

```ts
export const { styled } = createStitches({
  prefix: 'fluxui'
});
```
```tsx
export const StyledIndicator = styled('div', {
  color: '$gray50',
  width: 15,
  height: 15
});
```
```
Argument of type '{ color: "$gray50"; width: number; height: number; }' is not assignable to parameter of type 'RemoveIndex<CSS<{}, { borderWeights: { light: string; normal: string; bold: string; extrabold: string; black: string; }; borderWidths: { default: string; 0: string; 2: string; 4: string; 8: string; }; breakpoints: { ...; }; ... 9 more ...; zIndices: { ...; }; }, DefaultThemeMap, {}>> & ... 6 more ... & { ...; }'.
  Property 'length' is missing in type '{ color: "$gray50"; width: number; height: number; }' but required in type '{ toString: () => string; readonly length: string | number | {} | CSS<{}, { borderWeights: { light: string; normal: string; bold: string; extrabold: string; black: string; }; borderWidths: { default: string; 0: string; 2: string; 4: string; 8: string; }; ... 10 more ...; zIndices: { ...; }; }, DefaultThemeMap, {}> |...'.ts(2345)
lib.es5.d.ts(503, 14): 'length' is declared here.
```

## Expected behavior

Not to have to enter `length`.

## System information

- OS: Linux
- Version of Stitches: 1.2.6
- Version of Node.js: 17.4.0

### PR DESCRIPTION
**Pull request by jgoz**
Fixes #947.

TypeScript 4.6 changed... something... that necessitated checking `string extends Composers[K]` in addition to `Composers[K] extends string` to filter out string-assignable types.

This error actually surfaces in `test/build-types/built-types-test.tsx` if you switch to TS 4.6, but it looks like those types are generated manually based on published Stitches versions, so I didn't touch them in this PR.

### CODE REVIEW

**Review by csantos-nydig — APPROVED** [review]

**Review by ivanbanov — APPROVED** [review]

**Comment by jgoz** [comment]
@hadihallak 👋 friendly ping - this should unblock TypeScript 4.6 upgrades for Stitches users. If you had any time to take a look, we'd be grateful.

**Review by franklinjavier — APPROVED** [review]

**Comment by csantos-nydig** [comment]
> Thanks for submitting the PR.. will look into this and hopefuly merge it in the next few days

👋 sorry to bother @hadihallak - do you know when this might be available?

Thanks

**Comment by mario-jerkovic** [comment]
@hadihallak do you have ETA when this will land on main, and is there something I could do to speed this up 👍

Thank

**Comment by jgoz** [comment]
For anyone blocked on this, feel free to temporarily switch to [our fork](https://github.com/implydata/stitches/) `@implydata/stitches-react@^1.2.8` (or `@implydata/stitches-core`, etc.). The only difference with upstream is the inclusion of this PR. We fully intend to switch back once this becomes available in `@stitches/react`.

**Comment by csantos1113** [comment]
It's sad `stitches` feels unmaintained 😞 critical PRs like this one seating there open for 38 days

Thanks @jgoz for the fork

**Comment by lcoalves** [comment]
> It's sad `stitches` feels unmaintained disappointed critical PRs like this one seating there open for 38 days
> 
> Thanks @jgoz for the fork

This also makes me worried!

**Comment by Haugen** [comment]
Having the same issue here. Hope to see this getting some attention and merged soon!

**Comment by flex-jonghyen** [comment]
@hadihallak I cannot waiting to merge, Can you check this?

**Comment by hadihallak** [comment]
Sorry was OOO and didn't get to this. looking into this now

**Review by hadihallak — APPROVED** [review]
Thanks for the PR and apologies for the delay in merging it 🙏

**Comment by jansedlon** [comment]
Great, works even with TS 4.7!

**Pull request merged**
